momentumopt/src/momentumopt/setting/PlannerSetting.cpp

```diff
@@ -156,8 +156,29 @@ namespace momentumopt {
         YAML::ReadParameter(planner_vars, "reg_joint_position", reg_joint_position_);
         YAML::ReadParameter(planner_vars, "num_joint_viapoints", num_joint_viapoints_);
         YAML::ReadParameter(planner_vars, "num_base_viapoints", num_base_viapoints_);
+        // Kinematic paramters of second order IK (python)
+        YAML::ReadParameter(planner_vars, "swing_traj_via_z_second", swing_traj_via_z_second_);
+        YAML::ReadParameter(planner_vars, "w_lin_mom_tracking_second", w_lin_mom_tracking_second_);
+        YAML::ReadParameter(planner_vars, "w_ang_mom_tracking_second", w_ang_mom_tracking_second_);
+        YAML::ReadParameter(planner_vars, "w_endeff_contact_second", w_endeff_contact_second_);
+        YAML::ReadParameter(planner_vars, "w_endeff_tracking_second", w_endeff_tracking_second_);
+        YAML::ReadParameter(planner_vars, "p_endeff_tracking_second", p_endeff_tracking_second_);
+        YAML::ReadParameter(planner_vars, "p_com_tracking_second", p_com_tracking_second_);
+        YAML::ReadParameter(planner_vars, "w_joint_regularization_second", w_joint_regularization_second_);
+        YAML::ReadParameter(planner_vars, "d_endeff_tracking_second", d_endeff_tracking_second_);
+        YAML::ReadParameter(planner_vars, "p_orient_tracking_second", p_orient_tracking_second_);
+        YAML::ReadParameter(planner_vars, "d_orient_tracking_second", d_orient_tracking_second_);
+        YAML::ReadParameter(planner_vars, "p_joint_regularization_second", p_joint_regularization_second_);
+        YAML::ReadParameter(planner_vars, "d_joint_regularization_second", d_joint_regularization_second_);
+        YAML::ReadParameter(planner_vars, "mom_tracking_second", mom_tracking_second_);
+        YAML::ReadParameter(planner_vars, "num_joint_viapoints_second", num_joint_viapoints_second_);
+        YAML::ReadParameter(planner_vars, "num_base_viapoints_second", num_base_viapoints_second_);
+        YAML::ReadParameter(planner_vars, "use_second_order_inv_kin", use_second_order_inv_kin_); 
```

Comment:
Once we add these properties, we expect them to be specified in each yaml config file, even when we just use the first order optimization. I therefore wonder if we should/can do a check if the `use_second_order_inv_kin` exists and is toggled to true and only then try to read second order values? This way, we avoid updating all existing config files.

Reply:
and idem for the first order one